### PR TITLE
Add Alignment::Stretch and improve flex example

### PR DESCRIPTION
**Pull request by cmyr**
Alignment::Stretch forces children to occupy the maximum space on
the minor axis.

The more interesting part of this patch is that examples/flex
(previously examples/align) now allows the user to select the
settings for the demo widget, and updates in real-time; this is
useful for me for looking for evaluating layout issues.

Screen Shot 2020-03-09 at 1 37 13 PM
Screen Shot 2020-03-09 at 1 36 18 PM

### CODE REVIEW

**Comment by futurepaul** [comment]
This is so cool and useful! Is it possible to make the row view have a flexible height based on the window size so it's easier to see what's happening in those cases?

**Comment by cmyr** [comment]
good idea. I've added that, although it doesn't work as expected for two reasons: first off the align widgets keep things centered, and second off there's an issue in flex where it doesn't loosen incoming constraints before passing them on to children, so all children end up filling the space.

In any case I think that this example will be useful as we start patching these issues, it will be a good illustration of changes that get made.

Screen Shot 2020-03-09 at 2 42 03 PM

**Comment by cmyr** [comment]
(and it definitely helps highlight various problems)

**Comment by futurepaul** [comment]
I don't believe you've pushed this change yet?

Yeah I agree this is going to be super helpful identifying problems.

**Comment by cmyr** [comment]
indeed. updated. :)

File: druid/src/widget/flex.rs [hunk 1]
```diff
@@ -56,6 +57,11 @@ pub enum Alignment {
     /// In a vertical container, widgets are bottom aligned. In a horiziontal
     /// container, their trailing edges are aligned.
     End,
+    /// The widget is stretched to fill any available space on the minor axis.
```
[the remark this answers is not in the document]
Comment (cmyr) [reply druid/src/widget/flex.rs]: good catch

[the remark this answers is not in the document]
Comment (cmyr) [reply druid/src/widget/flex.rs]: I actually think there's a problem with my `Stretch` impl; basically it isn't using the size returned by the widget when painting, which could cause problems.

Flutter seems to throw an exception if you have a `Stretch` and the cross axis isn't bounded; I think for now I'm just going to remove this, and this PR can just be for the new demo.

[the remark this answers is not in the document]
Comment (futurepaul) [reply druid/src/widget/flex.rs]: Oh no!

**Pull request merged**